lib/MusicBrainz/Server/Controller/Account.pm

```diff
@@ -434,14 +433,13 @@ sub register : Path('/register') ForbiddenOnSlaves RequireSSL DenyWhenReadonly
         my $valid = 0;
         if ($use_captcha)
         {
-            my $challenge = $c->req->params->{recaptcha_challenge_field};
-            my $response = $c->req->params->{recaptcha_response_field};
+            my $response = $c->req->params->{'g-recaptcha-response'} // '';
 
-            $captcha_result = $captcha->check_answer(
+            $valid = $captcha->check_answer_v2(
                 DBDefs->RECAPTCHA_PRIVATE_KEY,
-                $c->req->address, $challenge, $response);
-
-            $valid = $captcha_result->{is_valid};
+                $response, $c->req->address
+                )->{is_valid}
+            unless $response eq '';
```

Comment:
No need to bother Google if the user didn’t submit any response at all (likely a non-JS browser or other client).